components/raftstore/src/store/peer_storage.rs

```diff
@@ -464,13 +464,13 @@ pub fn recover_from_applying_state<EK: KvEngine, ER: RaftEngine>(
 
     let raft_state = box_try!(engines.raft.get_raft_state(region_id)).unwrap_or_default();
 
-    // if we recv append log when applying snapshot, last_index in raft_local_state will
-    // larger than snapshot_index. since raft_local_state is written to raft engine, and
-    // raft write_batch is written after kv write_batch, raft_local_state may wrong if
+    // since raft_local_state is written to raft engine, and
+    // raft write_batch is written after kv write_batch. raft_local_state may wrong if
     // restart happen between the two write. so we copy raft_local_state to kv engine
-    // (snapshot_raft_state), and set snapshot_raft_state.last_index = snapshot_index.
-    // after restart, we need check last_index.
-    if last_index(&snapshot_raft_state) > last_index(&raft_state) {
+    // (snapshot_raft_state), and set snapshot_raft_state.hard_state.commit = snapshot_index.
```

Comment:
The comment is wrong, we don't set `hard_state.commit` explicitly.

Reply:
In `save_snapshot_raft_state_to`, we set commit of hard_state in snapshot_raft_state to snapshot_index. And then we persist snapshot_raft_state into kvdb.
```
let mut snapshot_raft_state = self.raft_state.clone();
        snapshot_raft_state
            .mut_hard_state()
            .set_commit(snapshot_index);
```

Reply:
Got it.